src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentationMeterOptions.cs

```diff
@@ -0,0 +1,70 @@
+// <copyright file="HttpClientInstrumentationMeterOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using System.Net;
+#if NETFRAMEWORK
+using System.Net.Http;
+#endif
+
+namespace OpenTelemetry.Instrumentation.Http;
+
+/// <summary>
+/// Options for HttpClient Meter instrumentation.
+/// </summary>
+public class HttpClientInstrumentationMeterOptions
+{
+    /// <summary>
+    /// Delegate for enrichment of recorded metric with additional tags.
+    /// </summary>
+    /// <param name="name">The name of the metric being enriched.</param>
+    /// <param name="requestMessage"><see cref="HttpRequestMessage"/>: the HttpRequestMessage object.</param>
+    /// <param name="tags"><see cref="TagList"/>: List of current tags. You can add additional tags to this list. </param>
+    public delegate void HttpRequestMessageEnrichmentFunc(string name, HttpRequestMessage requestMessage, ref TagList tags);
```

Comment:
I think we need to pass in both request and response.
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4098
@cijothomas - FYI

Reply:
I'll leave HttpResponse, if your changes appear earlier, I'll fix mine, and vice versa